mlx/traceable_base_directive.py

```diff
@@ -139,37 +145,48 @@ def process_options(self, node, options, docname=None):
 
                 | {
                 |     'target': { 'default': [''] },
-                |     'source': { 'default': '' },
-                |     'targettitle': { 'default': ['Target'], 'delimiter': ','},
+                |     'source': { 'default': '', 'is_pattern': True },
+                |     'targettitle': { 'default': ['Target'], 'delimiter': ',' },
                 |     'sourcetitle': { 'default': 'Source' },
                 |     'type': { 'default': [] },
                 | }
 
                 The 'default' configuration is mandatory. It is used to set the value in case none is given AND it is
                 used to determine the type of the option (list or single value).
-                The 'delimiter' value is optional and defaults to ' ' (single space). It is used to determine what
-                character needs to be used to split the option value into a list.
+                The 'delimiter' configuration is optional and defaults to ' ' (single space). It is used to determine
+                which character needs to be used to split the option value into a list.
+                The 'is_pattern' configuration is optional and defaults to False. If True, each option value is
+                treated as a regular expression pattern (str) and compiled to a re.Pattern object.
             docname (str): Document name.
 
         Returns:
             (bool) False if a required option is missing, True otherwise.
         """
         for option, option_config in options.items():
             if option in self.options:
+                value = self.options[option]
+                if option == 'filter':
+                    value = value.replace('\n', '').strip()
```

Comment:
This is usually better to have a space if you have a multi line without spaces? Strip will anyway cut out additional spaces on the end, but we still want to split the words.
```suggestion
                    value = value.replace('\n', ' ').strip()
```

Maybe we could split to more spaces to 1 using `sub` but I do not think that would be a problem

Reply:
No. In this case we really want to remove the newline characters. This new code replaces https://github.com/melexis/sphinx-traceability-extension/pull/293/files#diff-ceecbebd122c54fcd84d8cb602b2789ab703e5311625f509d0c970c5f197c5fcL63

This functionality was implemented in https://github.com/melexis/sphinx-traceability-extension/pull/206.